sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/aio/_text_analytics_client_async.py

```diff
@@ -606,14 +648,15 @@ async def begin_analyze_healthcare(  # type: ignore
         show_stats = kwargs.pop("show_stats", False)
         polling_interval = kwargs.pop("polling_interval", 5)
         continuation_token = kwargs.pop("continuation_token", None)
+        string_index_type = kwargs.pop("string_index_type", self._string_code_unit)
 
         doc_id_order = [doc.get("id") for doc in docs]
 
         try:
             return await self._client.begin_health(
                 docs,
                 model_version=model_version,
-                string_index_type=self._string_code_unit,
+                string_index_type=string_index_type,
```

Comment:
Can you add tests for strinig idnex type?
Basically make sure iin apii version 3.0, normal calls work, and that if you pass in striing nidex type. you get the error you're expecting
For api versioni 3.1 preview, make sure if you do't pass ini strinig index type we have Uniccode. And if you do pass in a string index type, we use that one insitead

Reply:
I've added a few more tests.  Let me know if you think anything is missing.